### PR TITLE
Add check for partition_keys when using append_date_dirs

### PR DESCRIPTION
**Pull request by CillianFn**
Related to #59

### CODE REVIEW

File: weather_dl/download_pipeline/parsers.py [hunk 1]
```diff
@@ -220,7 +220,8 @@ def _number_of_replacements(s: t.Text):
 
 
 def use_date_as_directory(config: t.Dict):
-    return 'date' in config['parameters']['partition_keys'] \
+    return 'partition_keys' in config['parameters'] \
+           and 'date' in config['parameters']['partition_keys'] \
```
Comment (alxmrs) [inline weather_dl/download_pipeline/parsers.py:224]: Nice catch!

(It's hard to see in this diff): Is there a test for a successful config with no partition keys? If not, can you add one?

File: weather_dl/download_pipeline/parsers_test.py [hunk 2]
```diff
@@ -565,6 +565,34 @@ def test_accepts_partition_keys_matching_sections(self):
...
+                    1990
+                    2000
+                    2010
+                    2020
+                """
+        ) as f:
+            config = process_config(f)
+            params = config.get('parameters', {})
+            self.assertTrue(bool(config))
+            self.assertIsInstance(params['partition_keys'], list)
```
Comment (CillianFn) [inline weather_dl/download_pipeline/parsers_test.py:594]: @alxmrs Checking that `partition_keys` is an instance of type list even when not specified in the config file seems to be intended logic [here](https://github.com/google/weather-tools/blob/main/weather_dl/download_pipeline/parsers.py#L309) & [here](https://github.com/google/weather-tools/blob/main/weather_dl/download_pipeline/parsers.py#L339-L340)

LMK if this still holds up

Comment (alxmrs) [reply weather_dl/download_pipeline/parsers_test.py:594]: I appreciate including this in the tests. However, this is more of an implementation detail and not part of the end interface – so no need to exercise it.

**Review by alxmrs — APPROVED** [review]

**Comment by CillianFn** [comment]
@alxmrs I accidentally deleted the branch, my bad.

**Comment by alxmrs** [comment]
Are you able to merge the change?

**Pull request merged**